addons/website_mass_mailing/static/src/js/website_mass_mailing.editor.js

```diff
@@ -54,13 +70,17 @@ options.registry.mailing_list_subscribe = options.Class.extend({
      * @see this.selectClass for parameters
      */
     toggleThanksButton(previewMode, widgetValue, params) {
-        const subscribeBtnEl = this.$target[0].querySelector('.js_subscribe_btn');
-        const thanksBtnEl = this.$target[0].querySelector('.js_subscribed_btn');
+        const toSubscribeEl =
+            this.$target[0].querySelector(".js_subscribe_btn, #newsletter_form");
+        const thanksMessageEl =
+            this.$target[0].querySelector(".js_subscribed_btn, .s_website_form_end_message");
 
-        thanksBtnEl.classList.toggle('o_disable_preview', !widgetValue);
-        thanksBtnEl.classList.toggle('o_enable_preview', widgetValue);
-        subscribeBtnEl.classList.toggle('o_enable_preview', !widgetValue);
-        subscribeBtnEl.classList.toggle('o_disable_preview', widgetValue);
+        thanksMessageEl.classList.toggle("o_enable_preview", widgetValue);
+        toSubscribeEl.classList.toggle("o_disable_preview", widgetValue);
+        if (this.$target[0].dataset.newsletterTemplate !== "form") {
```

Comment:
The if statement is not necessary here. Even if the two following lines are not usefull for the form, I would no treat the form as an exception here. In any case, these classes are removed before the save. Keep it simple IMO